lightwood/encoder/categorical/onehot.py

```diff
@@ -61,17 +70,16 @@ def prepare(self, priming_data, max_dimensions=20000):
 
             self._lang.removeWord(word_to_remove)
 
+        # Note: Is target assume that we are operating in "unknown_token" mode
         if self.is_target:
-            self.index_weights = [None] * self._lang.n_words
-            if self.target_class_distribution is not None:
-                self.index_weights[0] = np.mean(list(self.target_class_distribution.values()))
-            else:
-                self.index_weights[0] = 1
+            self.index_weights = [1] * self._lang.n_words
+            if self.target_weights is not None:
+                self.index_weights[0] = np.mean(list(self.target_weights.values()))
```

Comment:
Why do we do this?
```
                self.index_weights[0] = np.mean(list(self.target_weights.values()))
                self.target_weights[UNCOMMON_WORD] = self.index_weights[0]
```

This implies that the unknown weight is the average likelihood of any other class; set this to 0. We're surprisingly up-weighting unknowns.